### PR TITLE
[ENGA3-658]: Resolve Sonarcloud issues

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Resolve SonarCloud critical issues

Jira Ticket: [#658](https://opn-ooo.atlassian.net/browse/ENGA3-658)

#### 2. Description of change

Refactored a few functions to reduce cognitive complexity and replaced switch condition with alternative.

#### 3. Quality assurance

Check Sonarcloud after merge with master.

**🔧 Environments:**

- Platform version: Magento 2.4.5
- Omise plugin version: Omise-Magento 2.31.0
- PHP version: 8.1

### CODE REVIEW

File: Controller/Callback/Offsite.php [hunk 1]
```diff
@@ -106,18 +106,163 @@ public function execute()
...
+            $this->cancel($order, $e->getMessage());
+
+            return $this->redirect(self::PATH_CART);
+        }
+    }
+
+    /**
+     * Mark order as failed
+     *
+     * @param obejct $charge
```
Comment (danfowler) [inline Controller/Callback/Offsite.php:166]:
```suggestion
     * @param object $charge
```

File: Controller/Callback/Offsite.php [hunk 2]
```diff
@@ -106,18 +106,163 @@ public function execute()
...
+     * @param boolean $shopeePayFaild {TODO: Remove this once backend issue is fixed}
```
Comment (danfowler) [inline Controller/Callback/Offsite.php:167]:
```suggestion
     * @param boolean $shopeePayFailed {TODO: Remove this once backend issue is fixed}
```

File: Controller/Callback/Offsite.php [hunk 3]
```diff
@@ -106,18 +106,163 @@ public function execute()
...
+                $order->getBaseCurrency()->formatTxt($invoice->getBaseGrandTotal())
+            )
+        );
+
+        $order->save();
+        return $this->redirect(self::PATH_SUCCESS);
+    }
+
+    /**
+     * Mark order a pending
```
Comment (danfowler) [inline Controller/Callback/Offsite.php:216]:
```suggestion
     * Mark order as pending
```

File: Controller/Callback/Offsite.php [hunk 4]
```diff
@@ -106,18 +106,163 @@ public function execute()
...
+        // Update order state and status.
+        $order->setState(Order::STATE_PAYMENT_REVIEW);
+        $order->setStatus($order->getConfig()->getStateDefaultStatus(Order::STATE_PAYMENT_REVIEW));
+
+        // Add transaction.
+        $transaction = $payment->addTransaction(Transaction::TYPE_PAYMENT);
+        $transaction->setIsClosed(false);
+        $payment->addTransactionCommentsToOrder(
+            $transaction,
+            __('The payment has been processing.<br/>Due to the Bank process, this might takes a few seconds
```
Comment (danfowler) [inline Controller/Callback/Offsite.php:232]: The language here is not great. Can we have @muthuswamyopn look at it to improve the grammar?

File: Controller/Callback/Offsite.php [hunk 5]
```diff
@@ -106,18 +106,163 @@ public function execute()
...
+
+        $order->save();
+
+        // TODO: Should redirect users to a page that tell users that
+        // their payment is in review instead of success page.
+        return $this->redirect(self::PATH_SUCCESS);
+    }
+
+    /**
+     * Check if the transation is valid
```
Comment (danfowler) [inline Controller/Callback/Offsite.php:245]:
```suggestion
     * Check if the transaction is valid
```

File: Controller/Callback/Offsite.php (hunk 4 above)
Comment (aashishgurung) [reply Controller/Callback/Offsite.php:232]: Sure. I have DMed @muthuswamyopn.

**Review by ajzkk — APPROVED** [review]

**Review by danfowler — APPROVED** [review]

**Pull request merged**